File: alura/carreira_java/example30/src/br/com/flaviogf/Account.java
package br.com.flaviogf;

public class Account {
    private String number;
    private String agency;
    private double balance;

    public Account(String number, String agency, double balance) {
        this.number = number;
        this.agency = agency;
        this.balance = balance;
    }

    public double getBalance() {
        return balance;
    }

    public void deposit(double value) {
        balance += value;
    }

    public void withdraw(double value) {
        if (balance - value < 0) {
            throw new InsufficientFundsException(String.format("Account has %.2f and was tried to withdraw %.2f", balance, value));
        }

        balance -= value;
    }

    public void transfer(double value, Account to) {
        withdraw(value);
        to.deposit(value);
    }
}

File: alura/orientacao_a_objetos_com_php_002/bank/src/Models/Account.php
<?php

namespace Bank\Models;

class Account
{
    private Holder $holder;
    private float $balance;

    public function __construct(Holder $holder)
    {
        $this->holder = $holder;
        $this->balance = 0;
    }

    public function getBalance(): float
    {
        return $this->balance;
    }

    public function deposit(float $value): void
    {
        $this->balance += $value;
    }

    public function withdraw(float $value): void
    {
        $this->balance -= $value;
    }
}